logsmanagement/db_api.c

```diff
@@ -0,0 +1,1348 @@
+                        char filename[FILENAME_MAX + 1];
+                        snprintfz(filename, FILENAME_MAX, BLOB_STORE_FILENAME "%d", i);
+                        if(unlikely( 
+                                SQLITE_OK != (rc = sqlite3_bind_text(stmt_init_BLOBS_table, 1, filename, -1, NULL)) ||
+                                SQLITE_OK != (rc = sqlite3_bind_int64(stmt_init_BLOBS_table, 2, (sqlite3_int64) 0)) ||
+                                SQLITE_DONE != (rc = sqlite3_step(stmt_init_BLOBS_table)) ||
+                                SQLITE_OK != (rc = sqlite3_reset(stmt_init_BLOBS_table)))){
+                            throw_error(p_file_info->chart_name, ERR_TYPE_SQLITE, rc, __LINE__, __FILE__, __FUNCTION__);
+                            uv_mutex_unlock(p_file_info->db_mut);
+                            goto return_error;
+                        }
+                    }
+                    rc = sqlite3_finalize(stmt_init_BLOBS_table);
+                    do_sqlite_error_check(p_file_info, rc, SQLITE_OK);
+                }
+                rc = sqlite3_finalize(stmt_check_if_BLOBS_TABLE_exists);
+                do_sqlite_error_check(p_file_info, rc, SQLITE_OK);
+                
+                /* If LOGS_TABLE doesn't exist, create it */
+                rc = sqlite3_exec(p_file_info->db,
+                            "CREATE TABLE IF NOT EXISTS " LOGS_TABLE "("
+                            "Id                 INTEGER     PRIMARY KEY,"
+                            "FK_BLOB_Id         INTEGER     NOT NULL,"
+                            "BLOB_Offset        INTEGER     NOT NULL,"
+                            "Timestamp          INTEGER     NOT NULL,"
+                            "Msg_compr_size     INTEGER     NOT NULL,"
+                            "Msg_decompr_size   INTEGER     NOT NULL,"
+                            "Num_lines          INTEGER     NOT NULL,"
+                            "FOREIGN KEY (FK_BLOB_Id) REFERENCES " BLOBS_TABLE " (Id) ON DELETE CASCADE ON UPDATE CASCADE"
+                            ");",
+                            0, 0, &err_msg);
+                if (unlikely(SQLITE_OK != rc)) {
+                    collector_error("[%s]: Failed to create " LOGS_TABLE ", SQL error: %s", p_file_info->chart_name, err_msg);
+                    throw_error(p_file_info->chart_name, ERR_TYPE_SQLITE, rc, __LINE__, __FILE__, __FUNCTION__);
+                    uv_mutex_unlock(p_file_info->db_mut);
+                    goto return_error;
+                } else collector_info("[%s]: Table " LOGS_TABLE " created successfully", p_file_info->chart_name);
+                
+                /* Create index on LOGS_TABLE Timestamp
+                 * TODO: If this doesn't speed up queries, check SQLITE R*tree 
+                *        module. Requires benchmarking with/without index. */
+                rc = sqlite3_exec(p_file_info->db,
+                                    "CREATE INDEX IF NOT EXISTS logs_timestamps_idx "
```

Comment:
I suggest we always use timestamp index with `DESC` order.

Reply:
At this point only timestamp queries in ascending order are supported, I haven't added support yet for queries in descending timestamp order. Once I add that, I will change the timestamp index to `DESC` order.

Reply:
This is interesting. :thinking: .
We are real time monitoring software, so the logic is to take a look first in last values. 
Are you still only using ascendenting order?